Added Legion to Hard_Aggressive for Public Testing

Felnious opened the pull request:
### Work Done

1. Changed init.as to incorporate Legion when it is off and to disable it when turned off in the mod options (Legion Separation from armada and cortex unit configs. (Needed due to AI Crashing when legion is off).
2. Changed military.as to not turn Air Labs off till 999999 AA Threat has been achieved.
3. Changed factory.as to incorporate all Legion Labs.
4. Changed Commander.as to include Legion Factory Starts (May need tweaked for public use).
5. Changed Common.as to make Legion=Armada unless loaded through init.as.
6. Added base config files for Legion Separation from Cor/Arm.

### Armada/Cortex Changes
Added better Economy Progression in the Build_chain.json with more BP creation.
Changed units to be Aggressive ( Almost Suicidal). Threat and Power changes will be coming soon to make them more Effective.

Needs proper testing when ported over to main game as this is my first Pull Request. It works when I've tested.

rlcevg commented:
Test-run finished good, no crashes with and without legion faction.

**In my opinion PR is fine for squash-merge**: I see no importance in saving separate commits as it won't work if reverted in parts (except for the last one "Fixed Commander.AS for Normal Play").

Merging now allows testing AI behaviour in the wilds, though it does require clean up.

----

TODO for the future as per infolog.txt: Clean up the config.
- 1
```
<BARbarIAn-stable>: Load config: LuaRules/Configs/BARb/stable/config/hard_aggressive/factory_leg.json
<BARbarIAn-stable>: Config override: air_map
<BARbarIAn-stable>: Config override: map
<BARbarIAn-stable>: Config override: min_land
<BARbarIAn-stable>: Config override: no_air
<BARbarIAn-stable>: Config override: offset
<BARbarIAn-stable>: Config override: speed
<BARbarIAn-stable>: Config override: warn_probability
```
"Config override" means leaf object in json defined several times. It is OK to re-define the value but i bet the value is same in our case, because legion separation adds new things and does not re-define any common quotas.

- 2
```
<BARbarIAn-stable>: CONFIG hard_aggressive: has unknown UnitDef 'legplat'
<BARbarIAn-stable>: CONFIG hard_aggressive: has unknown UnitDef 'armapt3'
...
<BARbarIAn-stable>: CONFIG hard_aggressive: has unknown UnitDef 'corscavdrag_scav'
<BARbarIAn-stable>: CONFIG hard_aggressive: has unknown UnitDef 'legapt3'
<BARbarIAn-stable>: CONFIG hard_aggressive: has unknown UnitDef 'leggantuw'
<BARbarIAn-stable>: CONFIG hard_aggressive: leginfestor has unknown main role 'Builder'
<BARbarIAn-stable>: CONFIG hard_aggressive: has unknown UnitDef 'legmcluster'
<BARbarIAn-stable>: CONFIG hard_aggressive: has unknown UnitDef 'legministarfall'
<BARbarIAn-stable>: CONFIG hard_aggressive: has unknown UnitDef 'legplat'
<BARbarIAn-stable>: CONFIG hard_aggressive: has unknown UnitDef 'legrwall'
<BARbarIAn-stable>: CONFIG hard_aggressive: has unknown UnitDef 'legskirmtank'
...
<BARbarIAn-stable>: CONFIG hard_aggressive: has unknown UnitDef 'raptor_queen_veryeasy'
<BARbarIAn-stable>: CONFIG hard_aggressive: has unknown UnitDef 'raptor_queen_veryhard'
<BARbarIAn-stable>: CONFIG hard_aggressive: armalab's land_tier0 total probability = 1.320000
<BARbarIAn-stable>: CONFIG hard_aggressive: armalab's water_tier0 total probability = 1.420000
...
<BARbarIAn-stable>: CONFIG hard_aggressive: armalab's land_tier4 total probability = 1.710000
<BARbarIAn-stable>: CONFIG hard_aggressive: armalab's water_tier4 total probability = 2.280000
<BARbarIAn-stable>: CONFIG hard_aggressive: factory 'armasy' can't build 'armcarry'
<BARbarIAn-stable>: CONFIG hard_aggressive: armasy's air_tier1 total probability = 0.990000
<BARbarIAn-stable>: CONFIG hard_aggressive: armasy's land_tier1 total probability = 0.960000
<BARbarIAn-stable>: CONFIG hard_aggressive: armlab's air_tier0 total probability = 1.380000
<BARbarIAn-stable>: CONFIG hard_aggressive: armlab's land_tier0 total probability = 1.300000
...
<BARbarIAn-stable>: CONFIG hard_aggressive: armlab's air_tier3 total probability = 1.370000
<BARbarIAn-stable>: CONFIG hard_aggressive: armlab's land_tier3 total probability = 1.550000
<BARbarIAn-stable>: CONFIG hard_aggressive: factory 'coramsub' can't build 'corseal'
<BARbarIAn-stable>: CONFIG hard_aggressive: coramsub's land_tier0 total probability = 0.850000
<BARbarIAn-stable>: CONFIG hard_aggressive: coramsub's land_tier1 total probability = 0.060000
<BARbarIAn-stable>: CONFIG hard_aggressive: factory 'corasy' can't build 'corcarry'
<BARbarIAn-stable>: CONFIG hard_aggressive: corasy's air_tier1 total probability = 0.980000
<BARbarIAn-stable>: CONFIG hard_aggressive: corasy's land_tier1 total probability = 0.950000
<BARbarIAn-stable>: CONFIG hard_aggressive: factory 'corgantuw' can't build 'corseal'
<BARbarIAn-stable>: CONFIG hard_aggressive: corgantuw's land_tier0 total probability = 0.470000
<BARbarIAn-stable>: CONFIG hard_aggressive: legaap's land_tier0 total probability = 1.350000
...
<BARbarIAn-stable>: CONFIG hard_aggressive: legaap's land_tier5 total probability = 2.050000
<BARbarIAn-stable>: CONFIG hard_aggressive: factory 'legalab' can't build 'coramph'
<BARbarIAn-stable>: CONFIG hard_aggressive: legamsub's air_tier0 total probability = 0.500000
...
<BARbarIAn-stable>: CONFIG hard_aggressive: legamsub's water_tier1 total probability = 0.400000
<BARbarIAn-stable>: CONFIG hard_aggressive: legap's land_tier0 total probability = 1.350000
...
<BARbarIAn-stable>: CONFIG hard_aggressive: legap's land_tier5 total probability = 1.350000
<BARbarIAn-stable>: CONFIG hard_aggressive: legavp's land_tier0 total probability = 1.340000
...
<BARbarIAn-stable>: CONFIG hard_aggressive: leglab's land_tier3 total probability = 1.550000
<BARbarIAn-stable>: CONFIG hard_aggressive: has unknown UnitDef 'corptl'
<BARbarIAn-stable>: CONFIG hard_aggressive: has unknown UnitDef 'legfmd'
<BARbarIAn-stable>: CONFIG hard_aggressive: has unknown UnitDef 'legfdrag'
<BARbarIAn-stable>: CONFIG hard_aggressive: has unknown UnitDef 'corptl'
<BARbarIAn-stable>: CONFIG hard_aggressive: has unknown UnitDef 'armptl'
```
Unknown unitdefs just clutter infolog, while total units probabilities lead to wrong understanding of config (for maintainer). As unit with weight 0.5 is supposed to be selected for construction in 50% of cases. But if factory has 3 units and all have 0.5 the chance is actually 33%.
Yet if non-normalized weights are intended, then [warn_probability](https://github.com/Felnious/Beyond-All-ReasonFelnious/blob/master/luarules/configs/BARb/stable/config/hard_aggressive/factory.json#L16) should be `false`.

Felnious commented:
Yes I forgot to change the warn probability to false when I was merging. I've noticed that for air and for the bot lab, if you have the line in the factory json higher then 1.00 total, that they expand much fast with defenses and Mex control earlier, while simultaneously putting pressure on the players.

As for the cleanup, I will remove all duplicated within the configs. I tried it myself when you helped with the separation script and it felt off a bit, but could've just been a change that I made to make it feel sluggish. So I reverted back to having them.

Felnious commented:
Removed Duplicated Json Values in the main Configs:
Factory.json
Behaviour.json
Block_map.json
build_chain.json
economy.json

Fixed Commander.as close to how Original Barbs (with few changes).

Changed some Lab Values to have a better Composition within the Game. Mainly Legion/Armada Bot Labs.
Decreased Air Lab economy tiers to further boost T2 Chance.

As far as i am aware. This is ready to be "Squashed-Merged" Once Tested via @rlcevg

Felnious commented:
Recombined Block_map_Leg with Block_map due to them not merging under same values as @rlcevg has advised.
